src/util.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <print>
+#include <variant>
+namespace util {
+
+constexpr bool is_alpha(char c) noexcept { return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || c == '_'; }
+
+constexpr bool is_digit(char c) noexcept { return c >= '0' && c <= '9'; }
+
+constexpr bool is_alphanumeric(char c) noexcept { return is_alpha(c) || is_digit(c); }
+
+struct VariantVisitor
+{
+  void operator()(const std::monostate & /*unused*/) { std::print("_"); }
+
+  template<typename T> void operator()(const T &value) { std::print("{}", value); }
+};
```

Comment:
Refactor this:
1. Don't call it `VariantVisitor`: it's doing a very specific thing: printing the value of `T` (and assuming it's printable)
2. Remove the `std::monostate` overload

Also recommend reading on the `struct overloads` [pattern](https://en.cppreference.com/w/cpp/utility/variant/visit), which you *can* actually name `VariantVisitor`. [Here's](https://github.com/karnkaul/klib/blob/main/lib/include/klib/visitor.hpp) my own generic one, for example.